docs/style-spec/expressions.md

```diff
@@ -1,32 +1,34 @@
 # Style Function Expressions
 
 **Constants:**
-- LN2
-- PI
-- e
+- `[ "ln2" ]`
+- `[ "pi" ]`
+- `[ "e" ]`
 
 **Literal:**
 - JSON string / number / boolean literal
 
 **Property lookup:**
-- Feature property:  `{ "ref": "feature", "key": propertyName, "default": defaultValue}`
-- Map property:  `{"ref": "map",  "key": "zoom"|"pitch"|… }`
-  - Note: expressions that refer to the map zoom level are only evaluated at integer zoomslevels. When the map is at non-integer zoom levels, the expression's value will be approximated using linear or exponential interpolation.
+- Feature property: `[ "data", key_expr ]`
+- Map property:
+  - `[ "zoom" ]` (Note: expressions that refer to the map zoom level are only evaluated at integer zoomslevels. When the map is at non-integer zoom levels, the expression's value will be approximated using linear or exponential interpolation.)
+  - `[ "pitch" ]`
+  - etc.
 
 **Decision:**
 - `["if", boolean_expr, expr_if_true, expr_if_false]` 
-  - Alternative: `[` `"``if``"``, [ filter_expr, expr_if_true ], [ filter_expr, expr_if_true], …, expr_otherwise]` (where the first matching filter wins)
 
 **Boolean:**
-- has, !has
-- ==, !=
-- >, <, >=, <=, between(?),
-- in, !in
-- all, any, none
+- `[ "has", key_expr ]`
+- `[ "==", expr1, expr2]`
```

Comment:
> Or would we require downcase and a strip-diacritics function to be applied explicitly to either operand?

I’m not so sure that would be robust enough for internationalization purposes: https://github.com/mapbox/mapbox-gl-js/issues/4136#issuecomment-302550706.

Relatedly, should `==` take multiple arguments in order to say, for instance, `a == b == c`? (If not, that would open the door to using the second argument for folding flags.) If `==` shouldn’t take multiple arguments, should arithmetic operators like `*` take multiple arguments?